src/functionalTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/EnvelopeDeletionTest.java

```diff
@@ -74,25 +86,24 @@ public void should_keep_zip_file_after_failed_processing() throws Exception {
         uploadZipFile(srcZipFilename, destZipFilename); // invalid due to missing json file
 
         // ensure that processing has happened
-        await()
-            .atMost(scanDelay + 15_000, TimeUnit.MILLISECONDS)
-            .pollDelay(scanDelay * 2, TimeUnit.MILLISECONDS)
```

Comment:
`pollDelay` was the reason to wait for processing to happen. If you intend to sleep instead for X amount, why not sticking X amount to `pollDelay` function instead? Won't need a `TODO` either 🙃

Reply:
What we're effectively doing here is we wait for a specific amount of time and then make the assertion. This isn't really waiting until the condition is met, as you'd think by reading this chain. `Thread.sleep` makes it much more obvious. The TODO is about waiting for the processing of the file to happen (by polling some endpoint) and then making an assertion about the existence of the file. That would save us a lot of time on average.

Reply:
Ah, _endpoint polling-based solution_ 😉 . Probably would be enough. Still recommend using library instead of `Thread.sleep`

Reply:
Can be an endpoint or any other way of checking if the file has been processed.

The statement reads "await at most ... until", when in fact it's not like this at all. We're just waiting for a specific amount of time and then make an assertion. There's also a conflict between the values in `atMost` and `pollDelay`. Depending on the configuration, pollDelay may be greater than atMost, which leads to an error.